libcudacxx/include/cuda/stream_ref

```diff
@@ -108,9 +108,9 @@ public:
    * \param rhs The second `stream_view` to compare
    * \return true if equal, false if unequal
    */
-  _LIBCUDACXX_NODISCARD_ATTRIBUTE friend bool constexpr operator==(stream_ref __lhs, stream_ref __rhs) noexcept
+  _LIBCUDACXX_NODISCARD_ATTRIBUTE bool constexpr operator==(stream_ref __rhs) const noexcept
   {
-    return __lhs.__stream == __rhs.__stream;
+    return __stream == __rhs.__stream;
```

Comment:
I've just spent a while modifying the code in opposite direction because `const` was ignored in C++20 mode leading to nvcc complaining about "comparison is ambiguous in standard C++20 because the implied comparison operator with reversed parameters is an equally good match ".

Reply:
The issue here is different. NVCC 11.1 cannot handle the nodiscard attribute on a friend function

Reply:
Either make this conditionally nodiscard depending on NVCC version, or make the argument here also be taken by const ref, otherwise yeah we'll have issues in C++20, because it'll need to compare two overloads: `(stream_ref, const stream_ref &)` and `(const stream_ref &, stream_ref)` and there's just no rule that orders them.

Reply:
Done